### PR TITLE
concourse: Remove extra on_failure from template pipeline

### PR DESCRIPTION
**Pull request by doty-pivotal**
Recently the job level `ensure: <<set_failed` was added to all of the CCP jobs.  This pattern ensures that all CCP jobs will run that `set_failed` job at then end of any CCP based test.  In the case of a green, all tests destroy the cluster, and the `set_failed` runs but doesn't complain about the missing cluster.  In the case of any task going red, the job puts the cluster into a failed state for debugging.  Because of the ensure, all `on_failure: <<set_failed` are no longer needed. 

Additionally we added an bash newline continuation `\` to the end of the suggested set dev pipeline command.

- [ ] Backport to 5X_STABLE

Author: Nadeem Ghani <nghani@pivotal.io>
Author: Jim Doty <jdoty@pivotal.io>

### CODE REVIEW

**Comment by doty-pivotal** [comment]
@khuddlefish - we left the multi cluster DPM jobs alone here, but wouldn't mind sitting down with you and applying the pattern to your jobs, but we wanted to double check before doing that unilaterally.

File: concourse/pipelines/gpdb_master-generated.yml [hunk 1]
```diff
@@ -1229,8 +1223,6 @@ jobs:
       <<: *ccp_gen_cluster_default_params
       # platform is necessary to generate correct ssh config
       platform: sles12
-    on_failure:
-      <<: *ccp_destroy
```
Comment (doty-pivotal) [inline concourse/pipelines/gpdb_master-generated.yml]: The pattern of throwing away ccp clusters that fail the gen_cluster step is old, this removes a missed instance of that, as the cluster should get cleaned up like other clusters that go to the on_failure state do.  Now that CCP cluster generation is stable, any failure is actually interesting for debugging purposes.  @kmacoskey thoughts?

**Comment by khuddlefish** [comment]
@doty-pivotal Removing it from the multi-cluster jobs should be fine. Happy to pair with you if you would like assistance. Otherwise, looks good.

**Comment by kmacoskey** [comment]
I very much 100% agree. There is a bad habit being formed that `gen_cluster` failures are automatically infrastructure or CCP failures. Now that stability has been drastically improved, failures in gpinit during `gen_cluster` should be kept for debugging to determine why gpinit has failed, and if it's because of infrastructure flakiness there should be thought into improving resilience and at the very least improving output so that the output reflects exactly what the underlying flake is and what it's causing to fail.

**Comment by doty-pivotal** [comment]
We paired with @khuddlefish, to remove the unnecessary `destroy_cluster`'s and `on_failure`'s from the `gptransfer` jobs

We will backport this change to `5X_STABLE`.

**Pull request merged**